django_app/redbox_app/redbox_core/management/commands/delete_expired_data.py

```diff
@@ -51,3 +52,23 @@ def handle(self, *_args, **_kwargs):
                 counter += 1
 
         self.stdout.write(self.style.SUCCESS(f"Successfully deleted {counter} file objects"))
+
+        self.stdout.write(self.style.NOTICE(f"Deleting chats expired before {cutoff_date}"))
+        counter = 0
+        message_counter = 0
+
+        chats_to_delete = ChatHistory.objects.annotate(last_modified_at=Max("chatmessage__modified_at")).filter(
+            last_modified_at__lt=cutoff_date
+        )
+
```

Comment:
Is this to avoid looping through the chats_to_delete? (Because the message_counter is for the total ChatMessages not total ChatHistories).

Happy to just do chats_to_delete.delete() without a loop (and set `counter = chats_to_delete.count()`), but if so do we want to do these things currently in the loop or just avoid them:
* have a debug log for each ChatHistory deletion attempt (so we can easily track which have/haven't been done if something goes wrong)
* counting the ChatMessages as well as the ChatHistories deleted?

Reply:
>Is this to avoid looping through the chats_to_delete?

yes

I think we are less likely to get a failure if we delete them all in one go?

> counting the ChatMessages as well as the ChatHistories deleted?

I dont think it matter exactly what we count?